pkg/flow/internal/controller/node_component.go

```diff
@@ -284,12 +284,14 @@ func (cn *ComponentNode) evaluate(scope *vm.Scope) error {
 	if cn.managed == nil {
 		// We haven't built the managed component successfully yet.
 		managed, err := cn.reg.Build(cn.managedOpts, argsCopyValue)
-		if err != nil {
-			return fmt.Errorf("building component: %w", err)
-		}
 		cn.managed = managed
 		cn.args = argsCopyValue
 
+		// We do the error check at the end to allow the component to still return
+		// a component instance with an error to signal that it should still run.
+		if err != nil {
+			return fmt.Errorf("building component: %w", err)
```

Comment:
what do you think about logging this error in addition to returning it?

I am coming at this from the perspective of someone debugging the agent. We should provide as much telemetry data as possible for failed nodes. This is a small tradeoff for giving the extra flexibility in behavior (not stopping the agent on a single bad node)